### DISCUSSION TITLE
What does "add :can-create-file" mean?

### DISCUSSION THREAD

**Discussion by richardstevenhack** [Taitava/obsidian-shellcommands#399] [Q&A] [answered]
I'm trying to get a system command to output a date and create a file with that date

The shell command executes and outputs the date. But when I select "Open Files" I get popup that says the file doesn't exist and to "add :can-create-file to the shell command output". 

I don't know what that means or how to do it.

**Comment by Taitava** [comment]
Hi @richardstevenhack ! 🙂 

1. Will the date go to the name of the file, or to its content? I.e. is the name of the file a date? E.g. _2024-03-03.md_?
2. Can you tell the complete error message, including the file name that [Open files](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Open+files) tried to open, please? Maybe I could then get a better clue on what is going on.
3. Can you also give me the shell command you are executing, in case if it's not private, please? Perhaps I could then suggest some example modifications to the shell command to make it work with the _Open files_ output handler. 👍 

Generally, [Open files](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Open+files) expects that all output (to `stdout`, i.e. _standard output_ stream) indicates a file name/path (or multiple files, separated by line breaks).
- It then tries to open the outputted file name(s)/path(s) in Obsidian.
- If the outputted text is not a name/path to an existing file, the SC plugin gives the error message you encountered.
- `:can-create-file` can be added to the output text (trailing the file name/path) to allow creating a new file - **but from what I understood from your use-case, this is not desired**. Your system command has already created a file, and so there's no need to create any more files. Also, in case the output text happens to have mistakes, it's better that _Open files_ does not create incorrectly named files.

**Reply by richardstevenhack** [reply, marked as answer]
1) I create a new file with CTRL-N and it is called "Untitled". It exists in the Obsidian vault; I can see it in the Dolphin file explorer.
2) I run this command from the terminal:
`stat -c %w "/Data2/Planning/Obsidian Vault/Untitled.md`
3) It outputs:
`2024-03-03 13:07:28.184741408 -0800`
3) I run this command from shell-commands inside Obsidian:
`stat -c %w "/Data2/Planning/Obsidian Vault/Untitled.md` to look at the file. I'm not even outputting anything at this point.
4) I immediately get a popup that says:
`stat: cannot statx '/Data2/Planning/Obsidian Vault/Untitled.md': No such file or directory`

You're right that it's caused by the Flatpak environment. There is a way to override the Flatpak security. I already did allow all Flatpaks on my system to be able to access all user files using Flatseal. However, there  is a way to further loosen flatpak security to allow access to other areas such as the dbus system bus. 

To make this work, I had to change the permissions in Flatseal for Obsidian to allow access to "All System Files", "All System Libraries, executables and static data" and "All system configurations" and then reboot Obsidian. Now I can execute the stat command without getting the popup. This makes Flatpak much less secure when using Obsidian, but I can live with that in my situation.

I earlier found another way to get the date to input into a field in the front matter of the newly created file. I use the Templater plugin to execute the system command `date '+%Y-%m-%d %H:%M:%S` in the basic template I use when a new file is created. Since there is no file being accessed by this command, this works fine. So I am using this method rather than the stat command anyway. I could change the Obsidian Flatpak permissions back to more restricted, but I'll leave them loose in case I need to do something else which would be restricted.

So I think we can close this one out as solved.
